src/pageNavigation.js

```diff
@@ -147,18 +147,25 @@ async function addListener(listener, {
 
     // Compile the match patterns for the listener
     const matchPatternSet = matching.createMatchPatternSet(matchPatterns);
-    const contentScript = await browser.scripting.registerContentScripts([{
-        id: "pageNavigation",
+    const contentScriptId = "pageNavigation";
+    try {
+        await browser.scripting.unregisterContentScripts({
+            ids: [contentScriptId]
+        });
+    } catch (ex) {
+        console.debug(`Not unregistering content script ${contentScriptId}`, ex);
+    }
+    await browser.scripting.registerContentScripts([{
+        id: contentScriptId,
         js: ["dist/browser-polyfill.min.js", pageNavigationContentScript],
         matches: matchPatterns,
-        persistAcrossSessions: false,
+        persistAcrossSessions: true,
         runAt: "document_start"
     }]);
 
     // Store a record for the listener
     pageDataListeners.set(listener, {
         matchPatternSet,
-        contentScript,
```

Comment:
shouldn't we still unregister this content script if `removeListener` is called?

we do need to change `removeListener` to use `unregisterContentScripts` instead of `contentScript.unregister()`

Reply:
I'm not sure that this ever worked quite right in the first place... I think it would be ideal to unload content scripts when listeners are removed though, I'll see if I can get that working.

Reply:
OK I think the problem is that this is the only module that actually implements this; we should implement it consistently. A lot of this code is getting boilerplate at this point and should probably be consolidated somehow anyway, I'll file an issue.